src/modules/battery.cpp

```diff
@@ -275,24 +334,36 @@ const std::tuple<uint8_t, float, std::string, float> waybar::modules::Battery::g
     }
 
     return {cap, time_remaining, status, total_power / 1e6};
+#endif
   } catch (const std::exception& e) {
     spdlog::error("Battery: {}", e.what());
     return {0, 0, "Unknown", 0};
   }
 }
 
 const std::string waybar::modules::Battery::getAdapterStatus(uint8_t capacity) const {
+#if defined(__Linux__)
   if (!adapter_.empty()) {
     bool online;
     std::ifstream(adapter_ / "online") >> online;
+#else
```

Comment:
Cannot be `#else` here given `#include <sys/sysctl.h>` is hidden behind `#if defined(__FreeBSD__)`.
```suggestion
#elif defined(__linux__)
```